eng/code-quality-reports/src/main/java/com/azure/tools/checkstyle/checks/JavadocCodeSnippetCheck.java

```diff
@@ -0,0 +1,194 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.tools.checkstyle.checks;
+
+import com.puppycrawl.tools.checkstyle.api.DetailNode;
+import com.puppycrawl.tools.checkstyle.api.JavadocTokenTypes;
+import com.puppycrawl.tools.checkstyle.checks.javadoc.AbstractJavadocCheck;
+import com.puppycrawl.tools.checkstyle.utils.JavadocUtil;
+
+import java.util.regex.Pattern;
+
+/**
+ * Requirement of Javadoc annotation @codesnippet check:
+ *  (1) Use '{@codesnippet ...}' instead of '<code>', '<pre>'.
+ *  (2) No multiple lines span at the '{@code ...}'.
+ *  (3) naming pattern of codesnippet":
+ *    1, For package name, class name, they should all be lower case.
+ *       For method name,  the first letter of method name should be lower case.
+ *       For parameters, they should only be letters. Concatenate all parameters with dash character.
+ *    2, Jonathon will add more requirement specs later.
+ */
+public class JavadocCodeSnippetCheck extends AbstractJavadocCheck {
+
+    private static final String CLASS_PATH_REGEX = "[a-z.]+";
+    private static final String METHOD_NAME_REGEX = "^[a-z][a-zA-Z]+";
+    private static final String PARAMETERS_REGEX = "[a-zA-Z-]+";
```

Comment:
What is `Z-`? Are you trying to include the - symbol as part of the class set? It should be escaped. `\-`

Reply:
"[a-zA-Z\-]+" is not a valid regular expression in Java. I have tested "[a-zA-Z-]+" and "[a-zA-Z\\-]+" both works same.

Reply:
The first sentence is confusing. `[a-zA-Z-]+` _is_ a valid regular expression in Java.

I found the answer I was looking for [Regular Expression: Character classes](https://www.regular-expressions.info/charclass.html)